subsys/net/lib/sockets/mbedtls_dbg.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2021 Emil Lindqvist
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+#include "mbedtls_dbg.h"
+
+LOG_MODULE_REGISTER(mbedtls, CONFIG_MBEDTLS_DEBUG_LEVEL);
+
+static void dbg_func(void *ctx, int level, const char *file,
+		int line, const char *str)
+{
+	const char *p, *basename;
+
+	ARG_UNUSED(ctx);
+
+	if (!file || !str) {
+		return;
+	}
+
+	/* Extract basename from file */
+	for (p = basename = file; *p != '\0'; p++) {
+		if (*p == '/' || *p == '\\') {
+			basename = p + 1;
+		}
+	}
+
+#define LOG_ARGS "%s:%04d: %s", basename, line, log_strdup(str)
+	switch (level) {
+	case 1:
+		LOG_ERR(LOG_ARGS);
+		break;
+	case 2:
+		LOG_WRN(LOG_ARGS);
+		break;
+	case 3:
+		LOG_INF(LOG_ARGS);
+		break;
+	case 4:
+		LOG_DBG(LOG_ARGS);
+		break;
+	default:
+		LOG_ERR("mbedTLS invalid debug level (%d)", level);
+		break;
```

Comment:
I don't like this `LOG_ARGS` macros. What about using `Z_LOG()` and pass log level there? It is what `lvgl` glue code already uses.

Reply:
Why should we use some internal macro `Z_LOG()` here? The `LOG_xxx()` macro will call `Z_LOG()` anyway but we should not use those internal macros meant for logging subsystem.

Reply:
True, `z_` prefix denotes internal API. Using `Z_LOG()` allows to reduce the code, which is why I suggested that. Breaking `Z_LOG()` API can still be easily followed by updating this file, as long it is in the same repo. Each approach has of course pros and cons. Personally I would prefer to use `Z_LOG()` here, but this is just a slight preference, not a requirement.

Reply:
I think we should stick with the public `LOG_xxx` macros, the level variable maps conveniently to zephyr log levels so it could be used directly, but still I would use the public ones here.